scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/TaskResultCreator.java

```diff
@@ -79,48 +80,45 @@ public TaskResultImpl getTaskResult(SchedulerDBManager dbManager, InternalJob jo
-            etd = (EligibleTaskDescriptor) jobDescriptor.getPausedTasks().get(task.getId());
+            eligibleTaskDescriptor = (EligibleTaskDescriptor) jobDescriptor.getPausedTasks().get(task.getId());
         } else if (jobDescriptor.getRunningTasks().get(task.getId()) != null) {
-            etd = (EligibleTaskDescriptor) jobDescriptor.getRunningTasks().get(task.getId());
+            eligibleTaskDescriptor = (EligibleTaskDescriptor) jobDescriptor.getRunningTasks().get(task.getId());
         }
 
-        if (jobDescriptor.getRunningTasks().get(task.getId()) != null) {
-            taskResult = (TaskResultImpl) dbManager.loadTasksResults(job.getId(),
-                    Collections.singletonList(task.getId())).get(task.getId());
-        } else {
-            taskResult = getEmptyTaskResultWithTaskIdAndExecutionTime(task, exception, output);
-
-            if (etd != null) {
-                taskResult.setPropagatedVariables(getPropagatedVariables(dbManager, etd, job, task));
-            }
-        }
+        TaskResultImpl taskResult = getEmptyTaskResult(task, exception, output);
+        taskResult.setPropagatedVariables(getPropagatedVariables(dbManager, eligibleTaskDescriptor, job, task));
 
         return taskResult;
 
     }
 
-    public TaskResultImpl getEmptyTaskResultWithTaskIdAndExecutionTime(InternalTask task,
-                                                                       Throwable exception, TaskLogs output) {
+    public TaskResultImpl getEmptyTaskResult(InternalTask task, Throwable exception, TaskLogs output) {
         return new TaskResultImpl(task.getId(), exception,
                 output, System.currentTimeMillis() - task.getStartTime());
     }
 
     private Map<String, byte[]> getPropagatedVariables(SchedulerDBManager dbManager,
-                                                       EligibleTaskDescriptor etd, InternalJob job, InternalTask task) throws UnknownTaskException {
+            EligibleTaskDescriptor eligibleTaskDescriptor, InternalJob job, InternalTask task) throws UnknownTaskException {
+
         Map<String, byte[]> variables = new HashMap<>();
 
-        int numberOfParentTasks = etd.getParents().size();
-        if (job.getType() == JobType.TASKSFLOW) {
-            // retrieve from the database the previous task results if available
-            if ((numberOfParentTasks > 0) && task.handleResultsArguments()) {
-                variables = extractTaskResultsAndMergeIntoMap(dbManager, etd, job);
-            } else {
-                variables = extractJobVariables(job);
+        if (job.getType() == JobType.TASKSFLOW && eligibleTaskDescriptor != null) {
+            try{
+                TaskResultImpl taskResult = (TaskResultImpl) dbManager.loadTasksResults(job.getId(),
+                        Collections.singletonList(task.getId())).get(task.getId());
+                variables.putAll(taskResult.getPropagatedVariables());
+            } catch (DatabaseManagerException exception) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)